tools/export/gnuarmeclipse/__init__.py

```diff
@@ -244,7 +244,7 @@ def create_jinja_ctx(self):
             opts['ld']['system_libraries'] = self.system_libraries
             opts['ld']['script'] = join(id.capitalize(),
                                         "linker-script-%s.ld" % id)
-            opts['cpp_cmd'] = " ".join(toolchain.preproc)
+            opts['cpp_cmd'] = '"{}"'.format(toolchain.preproc[0]) + " " + " ".join(toolchain.preproc[1:])
```

Comment:
This might be more clear:
```python
opts['cpp_cmd'] = '"{}" {}'.format(toolchain.preproc[0]), " ".join(toolchain.preproc[1:])
```

Don't worry, it's semantically the same thing.